spec/unit/puppet/provider/registry_value_spec.rb

```diff
@@ -98,6 +97,30 @@
       reg_value.provider.destroy
       reg_value.provider.exists?.should be_false
     end
+
+    it "can destroy a randomly created REG_SZ value" do
+      create_and_destroy(path, :string, SecureRandom.uuid)
+    end
+
+    it "can destroy a randomly created REG_EXPAND_SZ value" do
+      create_and_destroy(path, :expand, "#{SecureRandom.uuid} system root is %SystemRoot%")
+    end
+
+    it "can destroy a randomly created REG_BINARY value" do
+      create_and_destroy(path, :binary, '01011010')
+    end
+
+    it "can destroy a randomly created REG_DWORD value" do
+      create_and_destroy(path, :dword, rand(2 ** 32 - 1))
+    end
+
+    it "can destroy a randomly created REG_QWORD value" do
+      create_and_destroy(path, :qword, rand(2 ** 64 - 1))
+    end
+
+    it "can destroy a randomly created REG_MULTI_SZ value" do
+      create_and_destroy(path, :array, [SecureRandom.uuid, SecureRandom.uuid])
+    end
```

Comment:
Minor nitpick - all of these touch the registry, so they should be instead in `spec/integration/puppet/provider/registry_value_spec.rb`